pkg/scaffold/cmd.go

```diff
@@ -66,13 +67,14 @@ func printVersion() {
 }
 
 func main() {
-	flag.Parse()
+	zapFactory := zap.FactoryForFlags(pflag.CommandLine)
```

Comment:
Should we maybe add a comment or two explaining why and what this is.

Reply:
As I was typing up the comment for the factory, I had a thought that we could eliminate it entirely.

Do we like this better?
```go
// Add the zap logger flag set to the CLI. The flag set
// must be added before calling pflag.Parse().
pflag.CommandLine.AddFlagSet(zap.FlagSet())
pflag.Parse()

// Use a zap logr.Logger implementation. If none of
// the zap flags are configured (or if the zap flag set
// is not being used), this defaults to a production
// zap logger.
//
// The logger instantiated here can be changed to any logger
// implementing the logr.Logger interface. This logger will
// be propagated through the whole operator, generating
// uniform and structured logs.
logf.SetLogger(zap.Logger())
```